rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -36,39 +36,40 @@ __gather_event_conditions(
   rmw_events_t * events,
   std::unordered_set<DDS::StatusCondition *> & status_conditions)
 {
-  RMW_CHECK_ARGUMENT_FOR_NULL(events, RMW_RET_INVALID_ARGUMENT);
-  std::unordered_map<DDS::StatusCondition *, DDS::StatusMask> status_mask_map;
-  // gather all status conditions and masks
-  for (size_t i = 0; i < events->event_count; ++i) {
-    auto current_event = static_cast<rmw_event_t *>(events->events[i]);
-    RMW_CHECK_ARGUMENT_FOR_NULL(current_event->data, RMW_RET_INVALID_ARGUMENT);
-    DDS::Entity * dds_entity = static_cast<ConnextCustomEventInfo *>(
-      current_event->data)->get_entity();
-    if (!dds_entity) {
-      RMW_SET_ERROR_MSG("Event handle is null");
-      return RMW_RET_ERROR;
-    }
-    DDS::StatusCondition * status_condition = dds_entity->get_statuscondition();
-    if (!status_condition) {
-      RMW_SET_ERROR_MSG("status condition handle is null");
-      return RMW_RET_ERROR;
+  if (events) {
```

Comment:
@ahcorde why this change?

Reply:
Events might be NULL [as defined in the API](https://github.com/ros2/rmw/blob/145612d50ead587e156fbd8515417c44108de6d3/rmw/include/rmw/rmw.h#L2038). Without this change this function will return RMW_RET_ERROR.